api/src/main/java/org/ehcache/exceptions/RethrowingCacheAccessException.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.exceptions;
+
+/**
+ * Generic exception used when an internal operation fails on a {@link org.ehcache.Cache} but shouldn't be
+ * handled by a resilience strategy but rather rethrown to the caller.
+ *
+ * @deprecated This mechanism is a stop-gap solution until {@link org.ehcache.resilience.ResilienceStrategy}
+ * instances can be plugged-in.
+ *
+ * @author Ludovic Orban
+ */
+@Deprecated
+public class RethrowingCacheAccessException extends CacheAccessException {
+
+  /**
+   * Create an instance of RethrowingCacheAccessException.
+   * @param cause the cause RuntimeException that will be rethrown.
+   */
+  public RethrowingCacheAccessException(RuntimeException cause) {
+    super(cause);
+  }
+
+  @Override
+  public synchronized RuntimeException getCause() {
```

Comment:
Thinking of it some more, no need for your impl to be synchronized since you call into super which will be (or not) depending on the VM running.